src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -0,0 +1,101 @@
+package frc.robot.subsystems;
+
+import java.util.function.BooleanSupplier;
+
+import com.revrobotics.CANSparkBase.ControlType;
+
+import SushiFrcLib.Motor.MotorConfig;
+import SushiFrcLib.Motor.MotorHelper;
+import SushiFrcLib.Sensors.absoluteEncoder.AbsoluteEncoder;
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+import com.revrobotics.CANSparkMax;
+
+public class Intake extends SubsystemBase {
+    private final CANSparkMax indexerMotor;
+    private final CANSparkMax rollerMotor;
+    private final CANSparkMax pivotMotor;
+    private static Intake instance;
+    private final ArmFeedforward intakeFeedforward;
+    private final AbsoluteEncoder absoluteEncoder;
+    private double pivotPos;
+
+    public static Intake getInstance() {
+        if (instance == null) {
+            return new Intake();
+        }
+        return instance;
+    }
+
+    private Intake() {
+        indexerMotor = Constants.Intake.INTAKE_INDEXER_CONFIG.createSparkMax();
+        rollerMotor = Constants.Intake.INTAKE_UPRIGHT_ROLLERS_CONFIG.createSparkMax();
+        pivotMotor = Constants.Intake.INTAKE_PIVOT_CONFIG.createSparkMax();
+        intakeFeedforward = new ArmFeedforward(0.0, Constants.Intake.G, 0.0);
+        absoluteEncoder = new AbsoluteEncoder(Constants.Intake.ENCODER_CHANNEL, Constants.Intake.ENCODER_ANGLE_OFFSET);
+        MotorHelper.setDegreeConversionFactor(pivotMotor, Constants.Intake.INTAKE_GEAR_RATIO);
+    }
+
+    public void resetPosition() {
+        pivotMotor.getEncoder().setPosition(getPosition());
+    }
+
+    public double getPosition() {
+        return absoluteEncoder.getPosition();
+    }
+
+    public BooleanSupplier closeToSetpoint(double setpoint) {
+        return () -> (getError(setpoint) < Constants.Intake.MAX_ERROR);
+    }
+
+    public double getAbsoluteError() {
```

Comment:
use getAbsolutePosition function here instead of absoulteEncoder.getNormalizedPosition